src/cdk/testing/dom-helpers.ts

```diff
@@ -0,0 +1,9 @@
+export function getTextWithExcludedElements(element: Element, excludeSelector: string) {
+  const clone = element.cloneNode(true) as Element;
+  const exclusions = clone.querySelectorAll(excludeSelector);
+  for (let i = 0; i < exclusions.length; i++) {
+    let child = exclusions[i];
+    child.parentNode?.removeChild(child);
+  }
+  return (clone.textContent || '').trim();
+}
```

Comment:
On this file name:

> Helper is a technical term meaning 'Bad abstraction'

😉 

How about `text-filtering.ts`?

Reply:
That works as long as its just this function here, but I asked her to move some other DOM related stuff here too

Reply:
I've renamed it to `text-filtering.ts` for now but will reorganize the other DOM scripts in a followup